nexus/reconfigurator/execution/src/lib.rs

```diff
@@ -123,143 +131,512 @@ pub async fn realize_blueprint_with_overrides(
+                    .into_iter()
+                    .map(|db_sled| {
+                        (
+                            SledUuid::from_untyped_uuid(db_sled.id()),
+                            Sled::from(db_sled),
+                        )
+                    })
+                    .collect();
+
+                StepSuccess::new(Arc::new(sleds_by_id)).into()
+            },
+        )
+        .register()
+}
+
+fn register_deploy_disks_step<'a>(
+    registrar: &ComponentRegistrar<'_, 'a>,
+    opctx: &'a OpContext,
+    blueprint: &'a Blueprint,
+    sleds: SharedStepHandle<Arc<BTreeMap<SledUuid, Sled>>>,
+) -> StepHandle<DeployDisksDone> {
+    registrar
+        .new_step(
+            ExecutionStepId::Ensure,
+            "Deploy physical disks",
+            move |cx| async move {
+                let sleds_by_id = sleds.into_value(cx.token()).await;
+                let done = omicron_physical_disks::deploy_disks(
+                    &opctx,
+                    &sleds_by_id,
+                    &blueprint.blueprint_disks,
+                )
+                .await
+                .map_err(merge_anyhow_list)?;
+
+                StepSuccess::new(done).into()
+            },
+        )
+        .register()
+}
 
-    // From this point on, we'll assume that any errors that we encounter do
-    // *not* require stopping execution.  We'll just accumulate them and return
-    // them all at the end.
+fn register_deploy_zones_step<'a>(
+    registrar: &ComponentRegistrar<'_, 'a>,
+    opctx: &'a OpContext,
+    blueprint: &'a Blueprint,
+    sleds: SharedStepHandle<Arc<BTreeMap<SledUuid, Sled>>>,
+) {
+    registrar
+        .new_step(
+            ExecutionStepId::Ensure,
+            "Deploy Omicron zones",
+            move |cx| async move {
+                let sleds_by_id = sleds.into_value(cx.token()).await;
+                omicron_zones::deploy_zones(
+                    &opctx,
+                    &sleds_by_id,
+                    &blueprint.blueprint_zones,
+                )
+                .await
+                .map_err(merge_anyhow_list)?;
+
+                StepSuccess::new(()).into()
+            },
+        )
+        .register();
+}
+
+fn register_plumb_firewall_rules_step<'a>(
+    registrar: &ComponentRegistrar<'_, 'a>,
+    opctx: &'a OpContext,
+    datastore: &'a DataStore,
+) {
+    // After deploying omicron zones, we may need to refresh OPTE service
+    // firewall rules. This is an idempotent operation, so we don't attempt
+    // to optimize out calling it in unnecessary cases, although it is only
+    // needed in cases where we've changed the set of services on one or more
+    // sleds, or the sleds have lost their firewall rules for some reason.
+    // Fixing the latter case is a side effect and should really be handled by a
+    // firewall-rule-specific RPW; once that RPW exists, we could trigger it
+    // here instead of pluming firewall rules ourselves.
```

Comment:
I realize this comment was moved verbatim, but while we're here: a firewall rule RPW does exist now to handle regularly ensuring sleds have firewall rules. Maybe make this

```suggestion
    // sleds.
    //
    // TODO-cleanup: We should trigger the `service-firewall-rules` RPW here
    // instead of doing this ourselves.
```

Reply:
Done, thanks